[bug] [macos] [v2] Static compilation error caused by compilation target being IOS.

### Describe the bug

I'm using the mlua project to embed the lua interpreter for my tauri, the `vendored` feature of mlua compiles the lua interpreter on-the-fly and embeds it into the application, the on-the-fly static compilation fails due to the tauri compilation target being `IOS`. Using cargo build will pass compilation, but similar to dev mode, you need to start the vite service.

### Reproduction

_No response_

### Expected behavior

Ability to customize the target platform for compilation, not just `IOS`. Or specify the target platform for compilation of a particular craate.

### Full `tauri info` output

```text
[✔] Environment
    - OS: Mac OS 15.0.0 arm64 (X64)
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.79.0 (129f3b996 2024-06-10)
    ✔ cargo: 1.79.0 (ffa9cf99a 2024-06-03)
    ✔ rustup: 1.27.1 (2024-04-24)
    ✔ Rust toolchain: 1.79.0-aarch64-apple-darwin (default)
    - node: 22.6.0
    - pnpm: 9.0.6
    - yarn: 1.22.22
    - npm: 10.8.2

[-] Packages
    - tauri 🦀: 2.0.0-rc.14
    - tauri-build 🦀: 2.0.0-rc.11
    - wry 🦀: 0.43.1
    - tao 🦀: 0.30.0
    - tauri-cli 🦀: 1.5.14
    - @tauri-apps/api : 2.0.0-rc.5
    - @tauri-apps/cli : 2.0.0-rc.16

[-] Plugins
    - tauri-plugin-shell 🦀: 2.0.0-rc.3
    - @tauri-apps/plugin-shell : 2.0.0-rc.1
    - tauri-plugin-http 🦀: 2.0.0-rc.5
    - @tauri-apps/plugin-http : 2.0.0-rc.2
    - tauri-plugin-fs 🦀: 2.0.0-rc.5
    - @tauri-apps/plugin-fs : 2.0.0-rc.2
    - tauri-plugin-dialog 🦀: 2.0.0-rc.7
    - @tauri-apps/plugin-dialog : 2.0.0-rc.1
    - tauri-plugin-store 🦀: 2.0.0-rc.3
    - @tauri-apps/plugin-store : 2.0.0-rc.1

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```


### Stack trace

```text
weed@MacBook-Pro src-tauri % npm run tauri build --bundles=app --target=aarch64-apple-darwin

> kanade-fofa@0.0.4 tauri
> tauri build

    Running beforeBuildCommand `npm run build`

> kanade-fofa@0.0.4 build
> vue-tsc --noEmit && vite build

vite v5.4.6 building for production...
✓ 3240 modules transformed.
dist/index.html                     0.46 kB │ gzip:     0.30 kB
dist/assets/index-C8_vPY4n.css      5.15 kB │ gzip:     1.73 kB
dist/assets/index-DAYD3FvN.js   3,564.34 kB │ gzip: 1,186.49 kB

(!) Some chunks are larger than 500 kB after minification. Consider:
- Using dynamic import() to code-split the application
- Use build.rollupOptions.output.manualChunks to improve chunking: https://rollupjs.org/configuration-options/#output-manualchunks
- Adjust chunk size limit for this warning via build.chunkSizeWarningLimit.
✓ built in 4.62s
   Compiling proc-macro2 v1.0.83
   Compiling unicode-ident v1.0.12
   Compiling libc v0.2.155
   Compiling serde v1.0.202
   Compiling autocfg v1.3.0
   Compiling cfg-if v1.0.0
   Compiling version_check v0.9.4
   Compiling parking_lot_core v0.9.10
   Compiling ppv-lite86 v0.2.17
   Compiling siphasher v0.3.11
   Compiling ahash v0.8.11
   Compiling shlex v1.3.0
   Compiling once_cell v1.19.0
   Compiling thiserror v1.0.61
   Compiling cc v1.1.20
   Compiling syn v1.0.109
   Compiling getrandom v0.1.16
   Compiling typenum v1.17.0
   Compiling lock_api v0.4.12
   Compiling generic-array v0.14.7
   Compiling zerocopy v0.7.35
   Compiling quote v1.0.36
   Compiling scopeguard v1.2.0
   Compiling syn v2.0.65
   Compiling memchr v2.7.2
   Compiling getrandom v0.2.15
   Compiling rand_core v0.6.4
   Compiling rand_core v0.5.1
   Compiling rand_pcg v0.2.1
   Compiling rand_chacha v0.3.1
   Compiling rand_chacha v0.2.2
   Compiling allocator-api2 v0.2.18
   Compiling itoa v1.0.11
   Compiling rand v0.8.5
   Compiling rand v0.7.3
   Compiling phf_shared v0.10.0
   Compiling phf_shared v0.8.0
   Compiling equivalent v1.0.1
   Compiling hashbrown v0.14.5
   Compiling ryu v1.0.18
   Compiling serde_json v1.0.117
   Compiling byteorder v1.5.0
   Compiling proc-macro-hack v0.5.20+deprecated
   Compiling phf_generator v0.8.0
   Compiling indexmap v2.2.6
   Compiling phf_generator v0.10.0
   Compiling tinyvec_macros v0.1.1
   Compiling string_cache_codegen v0.5.2
   Compiling phf_codegen v0.10.0
   Compiling tinyvec v1.6.0
   Compiling typeid v1.0.2
   Compiling percent-encoding v2.3.1
   Compiling unicode-bidi v0.3.15
   Compiling log v0.4.21
   Compiling markup5ever v0.11.0
   Compiling form_urlencoded v1.2.1
   Compiling fnv v1.0.7
   Compiling phf_codegen v0.8.0
   Compiling selectors v0.22.0
   Compiling unicode-normalization v0.1.23
   Compiling slab v0.4.9
   Compiling rustix v0.38.34
   Compiling core-foundation-sys v0.8.7
   Compiling ident_case v1.0.1
   Compiling strsim v0.11.1
   Compiling convert_case v0.4.0
   Compiling darling_core v0.20.9
   Compiling idna v0.5.0
   Compiling phf_shared v0.11.2
   Compiling winnow v0.5.40
   Compiling semver v1.0.23
   Compiling phf_generator v0.11.2
   Compiling crossbeam-utils v0.8.20
   Compiling new_debug_unreachable v1.0.6
   Compiling libm v0.2.8
   Compiling mac v0.1.1
   Compiling pin-project-lite v0.2.14
   Compiling precomputed-hash v0.1.1
   Compiling futf v0.1.5
   Compiling indexmap v1.9.3
   Compiling bitflags v1.3.2
   Compiling utf-8 v0.7.6
   Compiling dtoa v1.0.9
   Compiling serde_derive_internals v0.29.1
   Compiling tendril v0.4.3
   Compiling dtoa-short v0.3.4
   Compiling serde_derive v1.0.202
   Compiling thiserror-impl v1.0.61
   Compiling cssparser-macros v0.6.1
   Compiling phf_macros v0.8.0
   Compiling cssparser v0.27.2
   Compiling html5ever v0.26.0
   Compiling derive_more v0.99.17
   Compiling darling_macro v0.20.9
   Compiling phf_macros v0.11.2
   Compiling darling v0.20.9
   Compiling ctor v0.2.8
   Compiling phf v0.8.0
   Compiling serde_with_macros v3.8.1
   Compiling phf v0.10.1
   Compiling aho-corasick v1.1.3
   Compiling hashbrown v0.12.3
   Compiling regex-syntax v0.8.3
   Compiling num-conv v0.1.0
   Compiling alloc-no-stdlib v2.0.4
   Compiling unic-common v0.9.0
   Compiling matches v0.1.10
   Compiling camino v1.1.7
   Compiling futures-core v0.3.30
   Compiling nodrop v0.1.14
   Compiling unic-char-range v0.9.0
   Compiling stable_deref_trait v1.2.0
   Compiling itoa v0.4.8
   Compiling time-core v0.1.2
   Compiling servo_arc v0.1.1
   Compiling unic-char-property v0.9.0
   Compiling unic-ucd-version v0.9.0
   Compiling alloc-stdlib v0.2.2
   Compiling schemars_derive v0.8.20
   Compiling fxhash v0.2.1
   Compiling uuid v1.8.0
   Compiling regex-automata v0.4.6
   Compiling thin-slice v0.1.1
   Compiling const-oid v0.9.6
   Compiling schemars v0.8.20
   Compiling base64 v0.21.7
   Compiling cfb v0.7.3
   Compiling brotli-decompressor v4.0.1
   Compiling smallvec v1.13.2
   Compiling url v2.5.2
   Compiling serde_spanned v0.6.6
   Compiling toml_datetime v0.6.3
   Compiling bytes v1.6.0
   Compiling parking_lot v0.12.2
   Compiling toml_edit v0.20.2
   Compiling bitflags v2.5.0
   Compiling string_cache v0.8.7
   Compiling swift-rs v1.0.7
   Compiling jsonptr v0.4.7
   Compiling regex v1.10.6
   Compiling cargo-platform v0.1.8
   Compiling erased-serde v0.4.5
   Compiling unic-ucd-ident v0.9.0
   Compiling tracing-attributes v0.1.27
   Compiling toml v0.8.2
   Compiling anyhow v1.0.86
   Compiling dyn-clone v1.0.17
   Compiling same-file v1.0.6
   Compiling serde-untagged v0.1.6
   Compiling urlpattern v0.3.0
   Compiling walkdir v2.5.0
   Compiling kuchikiki v0.8.2
   Compiling cargo_metadata v0.18.1
   Compiling json-patch v2.0.0
   Compiling brotli v6.0.0
   Compiling serde_with v3.8.1
   Compiling infer v0.16.0
   Compiling phf v0.11.2
   Compiling futures-sink v0.3.30
   Compiling glob v0.3.1
   Compiling dunce v1.0.4
   Compiling subtle v2.6.1
   Compiling tokio-macros v2.2.0
   Compiling signal-hook-registry v1.4.2
   Compiling num_cpus v1.16.0
   Compiling socket2 v0.5.7
   Compiling mio v0.8.11
   Compiling native-tls v0.2.12
   Compiling block-buffer v0.10.4
   Compiling crypto-common v0.1.6
   Compiling tokio v1.37.0
   Compiling powerfmt v0.2.0
   Compiling digest v0.10.7
   Compiling deranged v0.3.11
   Compiling core-foundation v0.9.4
   Compiling tauri-utils v2.0.0-rc.11
   Compiling futures-macro v0.3.30
   Compiling spin v0.9.8
   Compiling time v0.3.36
   Compiling quick-xml v0.31.0
   Compiling objc_exception v0.1.2
   Compiling futures-io v0.3.30
   Compiling pin-utils v0.1.0
   Compiling line-wrap v0.2.0
   Compiling futures-task v0.3.30
   Compiling futures-util v0.3.30
   Compiling heck v0.5.0
   Compiling plist v1.6.1
   Compiling http v1.1.0
   Compiling tracing-core v0.1.32
   Compiling malloc_buf v0.0.6
   Compiling tracing v0.1.40
   Compiling objc v0.2.7
   Compiling rustc_version v0.4.0
   Compiling toml_edit v0.19.15
   Compiling zeroize v1.8.1
   Compiling option-ext v0.2.0
   Compiling dirs-sys v0.4.1
   Compiling embed-resource v2.4.2
   Compiling errno v0.3.9
   Compiling cpufeatures v0.2.12
   Compiling block v0.1.6
   Compiling dirs v5.0.1
   Compiling tauri-plugin v2.0.0-rc.11
   Compiling futures-channel v0.3.30
   Compiling cargo_toml v0.17.2
   Compiling foreign-types-macros v0.2.3
   Compiling security-framework-sys v2.11.1
   Compiling toml v0.7.8
   Compiling fastrand v2.1.1
   Compiling tauri-winres v0.1.1
   Compiling foreign-types-shared v0.3.1
   Compiling foreign-types v0.5.0
   Compiling tauri-build v2.0.0-rc.11
   Compiling tempfile v3.12.0
   Compiling security-framework v2.11.0
   Compiling sha2 v0.10.8
   Compiling num-traits v0.2.19
   Compiling httparse v1.8.0
   Compiling core-foundation v0.10.0
   Compiling ring v0.17.8
   Compiling pkg-config v0.3.30
   Compiling core-graphics-types v0.2.0
   Compiling tauri v2.0.0-rc.14
   Compiling either v1.12.0
   Compiling rustls-pki-types v1.8.0
   Compiling paste v1.0.15
   Compiling simd-adler32 v0.3.7
   Compiling base64 v0.22.1
   Compiling tokio-util v0.7.11
   Compiling http-body v1.0.0
   Compiling lazy_static v1.5.0
   Compiling pin-project-internal v1.1.5
   Compiling raw-window-handle v0.6.2
   Compiling try-lock v0.2.5
   Compiling objc-sys v0.3.5
   Compiling untrusted v0.9.0
   Compiling atomic-waker v1.1.2
   Compiling adler v1.0.2
   Compiling miniz_oxide v0.7.3
   Compiling h2 v0.4.6
   Compiling pin-project v1.1.5
   Compiling want v0.3.1
   Compiling cocoa-foundation v0.2.0
   Compiling core-graphics v0.24.0
   Compiling dpi v0.1.1
   Compiling time-macros v0.2.18
   Compiling crc32fast v1.4.2
   Compiling num-bigint-dig v0.8.4
   Compiling tower-layer v0.3.2
   Compiling tower-service v0.3.2
   Compiling rustls v0.22.4
   Compiling tower v0.4.13
   Compiling rustls-webpki v0.102.8
   Compiling flate2 v1.0.30
   Compiling hyper v1.3.1
   Compiling cocoa v0.26.0
   Compiling fdeflate v0.3.4
   Compiling objc_id v0.1.1
   Compiling core-graphics-types v0.1.3
   Compiling cookie v0.17.0
   Compiling vcpkg v0.2.15
   Compiling system-configuration-sys v0.5.0
   Compiling objc2-encode v4.0.3
   Compiling home v0.5.9
   Compiling base64ct v1.6.0
   Compiling libsqlite3-sys v0.30.1
   Compiling pem-rfc7468 v0.7.0
   Compiling objc2 v0.5.2
   Compiling png v0.17.13
   Compiling hyper-util v0.1.3
   Compiling crossbeam-channel v0.5.13
   Compiling serde_urlencoded v0.7.1
   Compiling idna v0.3.0
   Compiling tauri-runtime v2.0.0-rc.11
   Compiling psl-types v2.0.11
   Compiling wry v0.43.1
   Compiling minimal-lexical v0.2.1
   Compiling dispatch v0.2.0
   Compiling nom v7.1.3
   Compiling publicsuffix v2.2.3
   Compiling tokio-rustls v0.25.0
   Compiling ico v0.3.0
   Compiling objc2-foundation v0.2.2
   Compiling der v0.7.9
   Compiling core-graphics v0.23.2
   Compiling cocoa-foundation v0.1.2
   Compiling concurrent-queue v2.5.0
   Compiling http-body-util v0.1.1
   Compiling tokio-native-tls v0.3.1
   Compiling instant v0.1.13
   Compiling tauri-runtime-wry v2.0.0-rc.12
   Compiling parking v2.2.1
   Compiling unicode_categories v0.1.1
   Compiling unicode-segmentation v1.11.0
   Compiling crc-catalog v2.4.0
   Compiling sqlformat v0.2.6
   Compiling crc v3.2.1
   Compiling keyboard-types v0.7.0
   Compiling objc2-app-kit v0.2.2
   Compiling event-listener v5.3.1
   Compiling tao v0.30.0
   Compiling hyper-tls v0.6.0
   Compiling tokio-stream v0.1.16
   Compiling spki v0.7.3
   Compiling cocoa v0.25.0
   Compiling system-configuration v0.5.1
   Compiling cookie_store v0.20.0
   Compiling tauri-codegen v2.0.0-rc.11
   Compiling hyper-rustls v0.26.0
   Compiling crossbeam-queue v0.3.11
   Compiling rustls-pemfile v2.1.3
   Compiling atoi v2.0.0
   Compiling num-integer v0.1.46
   Compiling futures-intrusive v0.5.0
   Compiling webpki-roots v0.26.5
   Compiling tauri-plugin-fs v2.0.0-rc.5
   Compiling serialize-to-javascript-impl v0.1.1
   Compiling hashlink v0.9.1
   Compiling encoding_rs v0.8.34
   Compiling hex v0.4.3
   Compiling ipnet v2.9.0
   Compiling mime v0.3.17
   Compiling sync_wrapper v0.1.2
   Compiling reqwest v0.12.4
   Compiling sqlx-core v0.8.2
   Compiling serialize-to-javascript v0.1.1
   Compiling num-iter v0.1.45
   Compiling tauri-macros v2.0.0-rc.10
   Compiling window-vibrancy v0.5.0
   Compiling muda v0.14.1
   Compiling pkcs8 v0.10.2
   Compiling hmac v0.12.1
   Compiling serde_repr v0.1.19
   Compiling embed_plist v1.2.2
   Compiling state v0.6.0
   Compiling unicode-properties v0.1.2
   Compiling stringprep v0.1.5
   Compiling hkdf v0.12.4
   Compiling pkcs1 v0.7.5
   Compiling which v6.0.3
   Compiling signature v2.2.0
   Compiling md-5 v0.10.6
   Compiling dotenvy v0.15.7
   Compiling whoami v1.5.2
   Compiling rsa v0.9.6
   Compiling luajit-src v210.5.10+f725e44
   Compiling futures-executor v0.3.30
   Compiling sha1 v0.10.6
   Compiling flume v0.11.0
   Compiling lua-src v547.0.0
   Compiling sqlx-sqlite v0.8.2
   Compiling mlua-sys v0.6.3
   Compiling sqlx-mysql v0.8.2
   Compiling sqlx-postgres v0.8.2
   Compiling tauri-plugin-shell v2.0.0-rc.3
   Compiling tauri-plugin-dialog v2.0.0-rc.7
   Compiling tauri-plugin-http v2.0.0-rc.5
   Compiling tauri-plugin-store v2.0.0-rc.3
   Compiling rfd v0.14.1
   Compiling sqlx-macros-core v0.8.2
   Compiling objc-foundation v0.1.1
   Compiling ordered-float v2.10.1
   Compiling commoncrypto-sys v0.2.0
   Compiling pathdiff v0.2.1
   Compiling data-encoding v2.6.0
   Compiling tungstenite v0.23.0
   Compiling open v5.1.4
error: failed to run custom build command for `mlua-sys v0.6.3`

Caused by:
  process didn't exit successfully: `/Users/weed/program language/obj/rust/Kanade/src-tauri/target/release/build/mlua-sys-f22e9476b297e53a/build-script-main` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build
  OUT_DIR = Some(/Users/weed/program language/obj/rust/Kanade/src-tauri/target/release/build/mlua-sys-5738afa4bdfc3737/out)
  cargo:rerun-if-env-changed=CC_aarch64-apple-darwin
  CC_aarch64-apple-darwin = None
  cargo:rerun-if-env-changed=CC_aarch64_apple_darwin
  CC_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(false)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = Some(12.0)
  cargo:rerun-if-env-changed=CFLAGS_aarch64-apple-darwin
  CFLAGS_aarch64-apple-darwin = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_apple_darwin
  CFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  running cd "/Users/weed/program language/obj/rust/Kanade/src-tauri/target/release/build/mlua-sys-5738afa4bdfc3737/out/luajit-build/build/src" && BUILDMODE="static" STATIC_CC="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=12.0" TARGET_AR="/usr/bin/ar rcus" TARGET_LD="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=12.0" TARGET_STRIP="/usr/bin/strip" XCFLAGS="-fPIC" "make" "-e"
  HOSTCC    host/minilua.o
  HOSTLINK  host/minilua
  VERSION   luajit.h

  --- stderr
  clang: warning: using sysroot for 'MacOSX' but targeting 'iPhone' [-Wincompatible-sysroot]
  clang: warning: using sysroot for 'MacOSX' but targeting 'iPhone' [-Wincompatible-sysroot]
  ld: warning: <rdar://113405968> building for 'iOS', but linking in dylib (/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/lib/libSystem.B.tbd) built for 'macOS macCatalyst zippered(macOS/Catalyst)'
  ld: warning: <rdar://113405968> building for 'iOS', but linking in dylib (/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/lib/libSystem.B.tbd) built for 'macOS macCatalyst zippered(macOS/Catalyst)'
  make: *** [luajit.h] Killed: 9
  thread 'main' panicked at /Users/weed/.cargo/registry/src/index.crates.io-6f17d22bba15001f/luajit-src-210.5.10+f725e44/src/lib.rs:261:13:

  Error building LuaJIT:
      Command: cd "/Users/weed/program language/obj/rust/Kanade/src-tauri/target/release/build/mlua-sys-5738afa4bdfc3737/out/luajit-build/build/src" && BUILDMODE="static" STATIC_CC="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=12.0" TARGET_AR="/usr/bin/ar rcus" TARGET_LD="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=12.0" TARGET_STRIP="/usr/bin/strip" XCFLAGS="-fPIC" "make" "-e"
      Exit status: exit status: 2
      
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
failed to build app: failed to build app
    Error failed to build app: failed to build app
```


### Additional context

_No response_

> Would it be possible to create a minimal reproduction example? As far as i know we're not setting anything relevant in the cli so i'm a bit confused as to why `cargo build` works (which the cli calls internally).
> 
> Sidenote (shouldn't matter for the issue), when you're using npm you need another -- to tell it to forward the args to tauri, so `npm run tauri build -- --bundles=app --target=aarch64-apple-darwin`

As far as I've tested this doesn't happen on Windows x86_64 and Ubuntu X86_64, and I've used github actions to cross-compile, and only macos has this problem.

**attachments (you need to run `npm install`)**
[poc.zip](https://github.com/user-attachments/files/17052502/poc.zip)
minimal reproduction example:
```shell
weed@MacBook-Pro rust %npm create tauri-app@latest -- --rc
Need to install the following packages:
create-tauri-app@4.4.0
Ok to proceed? (y) y


> npx
> create-tauri-app --rc

✔ Project name · poc
✔ Identifier · com.poc.app
✔ Choose which language to use for your frontend · TypeScript / JavaScript - (pnpm, yarn, npm, bun)
✔ Choose your package manager · npm
✔ Choose your UI template · Vue - (https://vuejs.org/)
✔ Choose your UI flavor · TypeScript

Template created! To get started run:
  cd poc
  npm install
  npm run tauri android init
  npm run tauri ios init

For Desktop development, run:
  npm run tauri dev

For Android development, run:
  npm run tauri android dev

For iOS development, run:
  npm run tauri ios dev

weed@MacBook-Pro rust %cd poc
weed@MacBook-Pro poc %npm install
```
./src-tauri/Cargo.toml:
**enable `vendored` in mlua to ensure static compilation is used.**
```toml
[package]
name = "poc"
version = "0.1.0"
description = "A Tauri App"
authors = ["you"]
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[lib]
name = "poc_lib"
crate-type = ["lib", "cdylib", "staticlib"]

[build-dependencies]
tauri-build = { version = "2.0.0-rc", features = [] }

[dependencies]
tauri = { version = "2.0.0-rc", features = [] }
tauri-plugin-shell = "2.0.0-rc"
serde = { version = "1", features = ["derive"] }
serde_json = "1"
mlua = { version = "0.9.9", features = ["luajit", "vendored"] }
```
```shell
weed@MacBook-Pro poc %npm run tauri build 

> poc@0.1.0 tauri
> tauri build

    Running beforeBuildCommand `npm run build`

> poc@0.1.0 build
> vue-tsc --noEmit && vite build

vite v5.4.6 building for production...
✓ 19 modules transformed.
dist/index.html                  0.48 kB │ gzip:  0.31 kB
dist/assets/index-CSwEWTC4.css   1.76 kB │ gzip:  0.68 kB
dist/assets/index-MvvOLQM8.js   63.11 kB │ gzip: 25.28 kB
✓ built in 338ms
  Downloaded tauri-build v2.0.0-rc.12
  Downloaded tauri-codegen v2.0.0-rc.12
  Downloaded tauri-runtime v2.0.0-rc.12
  Downloaded tauri-macros v2.0.0-rc.11
  Downloaded tauri-runtime-wry v2.0.0-rc.13
  Downloaded bytes v1.7.2
  Downloaded anyhow v1.0.89
  Downloaded tauri-utils v2.0.0-rc.12
  Downloaded cc v1.1.21
  Downloaded unicode-segmentation v1.12.0
  Downloaded unicode-normalization v0.1.24
  Downloaded tauri v2.0.0-rc.15
  Downloaded rustix v0.38.37
  Downloaded 13 crates (1.2 MB) in 2m 17s
    Blocking waiting for file lock on package cache
   Compiling proc-macro2 v1.0.86
   Compiling unicode-ident v1.0.13
   Compiling libc v0.2.158
   Compiling cfg-if v1.0.0
   Compiling serde v1.0.210
   Compiling byteorder v1.5.0
   Compiling autocfg v1.3.0
   Compiling siphasher v0.3.11
   Compiling syn v1.0.109
   Compiling getrandom v0.1.16
   Compiling phf_shared v0.8.0
   Compiling phf_shared v0.10.0
   Compiling lock_api v0.4.12
   Compiling parking_lot_core v0.9.10
   Compiling memchr v2.7.4
   Compiling shlex v1.3.0
   Compiling proc-macro-hack v0.5.20+deprecated
   Compiling cc v1.1.21
   Compiling thiserror v1.0.63
   Compiling fnv v1.0.7
   Compiling typeid v1.0.2
   Compiling itoa v1.0.11
   Compiling quote v1.0.37
   Compiling equivalent v1.0.1
   Compiling strsim v0.11.1
   Compiling syn v2.0.77
   Compiling getrandom v0.2.15
   Compiling rand_core v0.6.4
   Compiling rand_core v0.5.1
   Compiling serde_json v1.0.128
   Compiling hashbrown v0.14.5
   Compiling ident_case v1.0.1
   Compiling smallvec v1.13.2
   Compiling rand_pcg v0.2.1
   Compiling ryu v1.0.18
   Compiling convert_case v0.4.0
   Compiling phf_shared v0.11.2
   Compiling winnow v0.5.40
   Compiling once_cell v1.19.0
   Compiling scopeguard v1.2.0
   Compiling semver v1.0.23
   Compiling tinyvec_macros v0.1.1
   Compiling new_debug_unreachable v1.0.6
   Compiling tinyvec v1.8.0
   Compiling indexmap v2.5.0
   Compiling mac v0.1.1
   Compiling core-foundation-sys v0.8.7
   Compiling precomputed-hash v0.1.1
   Compiling parking_lot v0.12.3
   Compiling futf v0.1.5
   Compiling indexmap v1.9.3
   Compiling utf-8 v0.7.6
   Compiling log v0.4.22
   Compiling bitflags v1.3.2
   Compiling dtoa v1.0.9
   Compiling percent-encoding v2.3.1
   Compiling unicode-normalization v0.1.24
   Compiling unicode-bidi v0.3.15
   Compiling form_urlencoded v1.2.1
   Compiling dtoa-short v0.3.5
   Compiling tendril v0.4.3
   Compiling idna v0.5.0
   Compiling aho-corasick v1.1.3
   Compiling phf v0.10.1
   Compiling itoa v0.4.8
   Compiling regex-syntax v0.8.4
   Compiling camino v1.1.9
   Compiling alloc-no-stdlib v2.0.4
   Compiling hashbrown v0.12.3
   Compiling darling_core v0.20.10
   Compiling serde_derive_internals v0.29.1
   Compiling unic-common v0.9.0
   Compiling matches v0.1.10
   Compiling unic-char-range v0.9.0
   Compiling nodrop v0.1.14
   Compiling stable_deref_trait v1.2.0
   Compiling unic-char-property v0.9.0
   Compiling servo_arc v0.1.1
   Compiling unic-ucd-version v0.9.0
   Compiling alloc-stdlib v0.2.2
   Compiling uuid v1.10.0
   Compiling fxhash v0.2.1
   Compiling thin-slice v0.1.1
   Compiling schemars v0.8.21
   Compiling brotli-decompressor v4.0.1
   Compiling unic-ucd-ident v0.9.0
   Compiling regex-automata v0.4.7
   Compiling cfb v0.7.3
   Compiling same-file v1.0.6
   Compiling dyn-clone v1.0.17
   Compiling serde_derive v1.0.210
   Compiling zerocopy-derive v0.7.35
   Compiling thiserror-impl v1.0.63
   Compiling cssparser-macros v0.6.1
   Compiling cssparser v0.27.2
   Compiling html5ever v0.26.0
   Compiling darling_macro v0.20.10
   Compiling derive_more v0.99.18
   Compiling darling v0.20.10
   Compiling zerocopy v0.7.35
   Compiling serde_with_macros v3.9.0
   Compiling ppv-lite86 v0.2.20
   Compiling ctor v0.2.8
   Compiling rand_chacha v0.3.1
   Compiling rand_chacha v0.2.2
   Compiling rand v0.8.5
   Compiling rand v0.7.3
   Compiling schemars_derive v0.8.21
   Compiling regex v1.10.6
   Compiling anyhow v1.0.89
   Compiling base64 v0.21.7
   Compiling infer v0.16.0
   Compiling phf_generator v0.8.0
   Compiling phf_generator v0.10.0
   Compiling phf_codegen v0.8.0
   Compiling phf_generator v0.11.2
   Compiling phf_codegen v0.10.0
   Compiling string_cache_codegen v0.5.2
   Compiling phf_macros v0.8.0
   Compiling selectors v0.22.0
   Compiling phf_macros v0.11.2
   Compiling brotli v6.0.0
   Compiling walkdir v2.5.0
   Compiling dunce v1.0.5
   Compiling glob v0.3.1
   Compiling markup5ever v0.11.0
   Compiling pin-project-lite v0.2.14
   Compiling objc_exception v0.1.2
   Compiling phf v0.8.0
   Compiling phf v0.11.2
   Compiling futures-core v0.3.30
   Compiling slab v0.4.9
   Compiling typenum v1.17.0
   Compiling version_check v0.9.5
   Compiling powerfmt v0.2.0
   Compiling deranged v0.3.11
   Compiling generic-array v0.14.7
   Compiling core-foundation v0.10.0
   Compiling serde_spanned v0.6.7
   Compiling toml_datetime v0.6.3
   Compiling string_cache v0.8.7
   Compiling toml_edit v0.20.2
   Compiling bitflags v2.6.0
   Compiling bytes v1.7.2
   Compiling url v2.5.2
   Compiling cargo-platform v0.1.8
   Compiling erased-serde v0.4.5
   Compiling jsonptr v0.4.7
   Compiling json-patch v2.0.0
   Compiling swift-rs v1.0.7
   Compiling serde-untagged v0.1.6
   Compiling urlpattern v0.3.0
   Compiling cargo_metadata v0.18.1
   Compiling serde_with v3.9.0
   Compiling http v1.1.0
   Compiling kuchikiki v0.8.2
   Compiling toml v0.8.2
   Compiling rustc_version v0.4.1
   Compiling toml_edit v0.19.15
   Compiling socket2 v0.5.7
   Compiling mio v1.0.2
   Compiling malloc_buf v0.0.6
   Compiling option-ext v0.2.0
   Compiling num-conv v0.1.0
   Compiling tauri-utils v2.0.0-rc.12
   Compiling time-core v0.1.2
   Compiling time v0.3.36
   Compiling tokio v1.40.0
   Compiling dirs-sys v0.4.1
   Compiling objc v0.2.7
   Compiling toml v0.7.8
   Compiling embed-resource v2.4.3
   Compiling core-graphics-types v0.2.0
   Compiling foreign-types-macros v0.2.3
   Compiling futures-macro v0.3.30
   Compiling quick-xml v0.32.0
   Compiling base64 v0.22.1
   Compiling pin-utils v0.1.0
   Compiling objc-sys v0.3.5
   Compiling heck v0.5.0
   Compiling foreign-types-shared v0.3.1
   Compiling futures-task v0.3.30
   Compiling rustix v0.38.37
   Compiling plist v1.7.0
   Compiling futures-util v0.3.30
   Compiling foreign-types v0.5.0
   Compiling tauri-winres v0.1.1
   Compiling dirs v5.0.1
   Compiling cargo_toml v0.17.2
   Compiling errno v0.3.9
   Compiling simd-adler32 v0.3.7
   Compiling block v0.1.6
   Compiling crossbeam-utils v0.8.20
   Compiling adler2 v2.0.0
   Compiling httparse v1.9.4
   Compiling miniz_oxide v0.8.0
   Compiling cocoa-foundation v0.2.0
   Compiling tauri-build v2.0.0-rc.12
   Compiling core-graphics v0.24.0
   Compiling http-body v1.0.1
   Compiling dpi v0.1.1
   Compiling pin-project-internal v1.1.5
   Compiling security-framework-sys v2.11.1
   Compiling core-foundation v0.9.4
   Compiling crc32fast v1.4.2
   Compiling adler v1.0.2
   Compiling fastrand v2.1.1
   Compiling try-lock v0.2.5
   Compiling raw-window-handle v0.6.2
   Compiling objc2-encode v4.0.3
   Compiling native-tls v0.2.12
   Compiling objc2 v0.5.2
   Compiling pin-project v1.1.5
   Compiling want v0.3.1
   Compiling tempfile v3.12.0
   Compiling flate2 v1.0.33
   Compiling security-framework v2.11.1
   Compiling miniz_oxide v0.7.4
   Compiling cocoa v0.26.0
   Compiling fdeflate v0.3.4
   Compiling crypto-common v0.1.6
   Compiling block-buffer v0.10.4
   Compiling futures-channel v0.3.30
   Compiling tracing-core v0.1.32
   Compiling tower-service v0.3.3
   Compiling either v1.13.0
   Compiling home v0.5.9
   Compiling tower-layer v0.3.3
   Compiling tower v0.4.13
   Compiling which v6.0.3
   Compiling tracing v0.1.40
   Compiling hyper v1.4.1
   Compiling digest v0.10.7
   Compiling png v0.17.13
   Compiling crossbeam-channel v0.5.13
   Compiling objc2-foundation v0.2.2
   Compiling tauri v2.0.0-rc.15
   Compiling cpufeatures v0.2.14
   Compiling wry v0.43.1
   Compiling tauri-runtime v2.0.0-rc.12
   Compiling objc2-app-kit v0.2.2
   Compiling sha2 v0.10.8
   Compiling tokio-native-tls v0.3.1
   Compiling ico v0.3.0
   Compiling hyper-util v0.1.8
   Compiling luajit-src v210.5.10+f725e44
   Compiling http-body-util v0.1.2
   Compiling tauri-plugin v2.0.0-rc.12
   Compiling objc_id v0.1.1
   Compiling lua-src v547.0.0
   Compiling instant v0.1.13
   Compiling tauri-runtime-wry v2.0.0-rc.13
   Compiling pkg-config v0.3.30
   Compiling futures-sink v0.3.30
   Compiling lazy_static v1.5.0
   Compiling unicode-segmentation v1.12.0
   Compiling dispatch v0.2.0
   Compiling rustls-pki-types v1.8.0
   Compiling tao v0.30.0
   Compiling mlua-sys v0.6.3
   Compiling keyboard-types v0.7.0
   Compiling rustls-pemfile v2.1.3
   Compiling tokio-util v0.7.12
   Compiling hyper-tls v0.6.0
   Compiling tauri-plugin-shell v2.0.0-rc.3
   Compiling tauri-codegen v2.0.0-rc.12
   Compiling serde_urlencoded v0.7.1
   Compiling sync_wrapper v1.0.1
   Compiling serialize-to-javascript-impl v0.1.1
   Compiling num-traits v0.2.19
   Compiling ipnet v2.10.0
   Compiling mime v0.3.17
   Compiling tauri-macros v2.0.0-rc.11
   Compiling reqwest v0.12.7
   Compiling serialize-to-javascript v0.1.1
   Compiling muda v0.14.1
   Compiling window-vibrancy v0.5.2
   Compiling serde_repr v0.1.19
   Compiling pathdiff v0.2.1
   Compiling embed_plist v1.2.2
   Compiling state v0.6.0
   Compiling open v5.3.0
error: failed to run custom build command for `mlua-sys v0.6.3`

Caused by:
  process didn't exit successfully: `/Users/weed/program language/obj/rust/poc/src-tauri/target/release/build/mlua-sys-b2606ce350961ae1/build-script-main` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=build
  OUT_DIR = Some(/Users/weed/program language/obj/rust/poc/src-tauri/target/release/build/mlua-sys-25ae8c09461486ca/out)
  cargo:rerun-if-env-changed=CC_aarch64-apple-darwin
  CC_aarch64-apple-darwin = None
  cargo:rerun-if-env-changed=CC_aarch64_apple_darwin
  CC_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CC
  HOST_CC = None
  cargo:rerun-if-env-changed=CC
  CC = None
  cargo:rerun-if-env-changed=CC_ENABLE_DEBUG_OUTPUT
  RUSTC_WRAPPER = None
  cargo:rerun-if-env-changed=CRATE_CC_NO_DEFAULTS
  CRATE_CC_NO_DEFAULTS = None
  DEBUG = Some(false)
  cargo:rerun-if-env-changed=MACOSX_DEPLOYMENT_TARGET
  MACOSX_DEPLOYMENT_TARGET = Some(10.13)
  cargo:rerun-if-env-changed=CFLAGS_aarch64-apple-darwin
  CFLAGS_aarch64-apple-darwin = None
  cargo:rerun-if-env-changed=CFLAGS_aarch64_apple_darwin
  CFLAGS_aarch64_apple_darwin = None
  cargo:rerun-if-env-changed=HOST_CFLAGS
  HOST_CFLAGS = None
  cargo:rerun-if-env-changed=CFLAGS
  CFLAGS = None
  running cd "/Users/weed/program language/obj/rust/poc/src-tauri/target/release/build/mlua-sys-25ae8c09461486ca/out/luajit-build/build/src" && BUILDMODE="static" STATIC_CC="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=10.13" TARGET_AR="/usr/bin/ar rcus" TARGET_LD="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=10.13" TARGET_STRIP="/usr/bin/strip" XCFLAGS="-fPIC" "make" "-e"
  HOSTCC    host/minilua.o
  HOSTLINK  host/minilua
  VERSION   luajit.h

  --- stderr
  clang: warning: using sysroot for 'MacOSX' but targeting 'iPhone' [-Wincompatible-sysroot]
  clang: warning: using sysroot for 'MacOSX' but targeting 'iPhone' [-Wincompatible-sysroot]
  ld: warning: <rdar://113405968> building for 'iOS', but linking in dylib (/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/lib/libSystem.B.tbd) built for 'macOS macCatalyst zippered(macOS/Catalyst)'
  ld: warning: <rdar://113405968> building for 'iOS', but linking in dylib (/Library/Developer/CommandLineTools/SDKs/MacOSX15.0.sdk/usr/lib/libSystem.B.tbd) built for 'macOS macCatalyst zippered(macOS/Catalyst)'
  make: *** [luajit.h] Killed: 9
  thread 'main' panicked at /Users/weed/.cargo/registry/src/index.crates.io-6f17d22bba15001f/luajit-src-210.5.10+f725e44/src/lib.rs:261:13:

  Error building LuaJIT:
      Command: cd "/Users/weed/program language/obj/rust/poc/src-tauri/target/release/build/mlua-sys-25ae8c09461486ca/out/luajit-build/build/src" && BUILDMODE="static" STATIC_CC="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=10.13" TARGET_AR="/usr/bin/ar rcus" TARGET_LD="/usr/bin/cc -O2 -ffunction-sections -fdata-sections -fPIC --target=arm64-apple-darwin -mmacosx-version-min=10.13" TARGET_STRIP="/usr/bin/strip" XCFLAGS="-fPIC" "make" "-e"
      Exit status: exit status: 2

  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
warning: build failed, waiting for other jobs to finish...
failed to build app: failed to build app
    Error failed to build app: failed to build app
```
when i use `cargo build`:
```shell
weed@MacBook-Pro poc %cd src-tauri
weed@MacBook-Pro src-tauri %cargo build
   Compiling cfg-if v1.0.0
   Compiling byteorder v1.5.0
   Compiling siphasher v0.3.11
   Compiling memchr v2.7.4
   Compiling itoa v1.0.11
   Compiling libc v0.2.158
   Compiling serde v1.0.210
   Compiling fnv v1.0.7
   Compiling getrandom v0.2.15
   Compiling getrandom v0.1.16
   Compiling ryu v1.0.18
   Compiling zerocopy v0.7.35
   Compiling once_cell v1.19.0
   Compiling rand_core v0.6.4
   Compiling rand_core v0.5.1
   Compiling phf_shared v0.10.0
   Compiling phf_shared v0.8.0
   Compiling equivalent v1.0.1
   Compiling rand_pcg v0.2.1
   Compiling hashbrown v0.14.5
   Compiling new_debug_unreachable v1.0.6
   Compiling tinyvec_macros v0.1.1
   Compiling log v0.4.22
   Compiling precomputed-hash v0.1.1
   Compiling tinyvec v1.8.0
   Compiling winnow v0.5.40
   Compiling percent-encoding v2.3.1
   Compiling unicode-bidi v0.3.15
   Compiling mac v0.1.1
   Compiling ppv-lite86 v0.2.20
   Compiling form_urlencoded v1.2.1
   Compiling futf v0.1.5
   Compiling indexmap v2.5.0
   Compiling dtoa v1.0.9
   Compiling rand_chacha v0.3.1
   Compiling rand_chacha v0.2.2
   Compiling utf-8 v0.7.6
   Compiling rand v0.8.5
   Compiling rand v0.7.3
   Compiling unicode-normalization v0.1.24
   Compiling bitflags v1.3.2
   Compiling tendril v0.4.3
   Compiling dtoa-short v0.3.5
   Compiling phf v0.10.1
   Compiling matches v0.1.10
   Compiling itoa v0.4.8
   Compiling darling_core v0.20.10
   Compiling phf_generator v0.10.0
   Compiling idna v0.5.0
   Compiling thiserror v1.0.63
   Compiling phf_codegen v0.10.0
   Compiling string_cache_codegen v0.5.2
   Compiling phf_generator v0.8.0
   Compiling nodrop v0.1.14
   Compiling phf_codegen v0.8.0
   Compiling stable_deref_trait v1.2.0
   Compiling hashbrown v0.12.3
   Compiling phf_macros v0.8.0
   Compiling serde_json v1.0.128
   Compiling selectors v0.22.0
   Compiling servo_arc v0.1.1
   Compiling typeid v1.0.2
   Compiling markup5ever v0.11.0
   Compiling aho-corasick v1.1.3
   Compiling fxhash v0.2.1
   Compiling serde_spanned v0.6.7
   Compiling toml_datetime v0.6.3
   Compiling unic-char-range v0.9.0
   Compiling unic-common v0.9.0
   Compiling regex-syntax v0.8.4
   Compiling phf v0.8.0
   Compiling thin-slice v0.1.1
   Compiling unic-ucd-version v0.9.0
   Compiling unic-char-property v0.9.0
   Compiling phf_shared v0.11.2
   Compiling alloc-no-stdlib v2.0.4
   Compiling lock_api v0.4.12
   Compiling phf_generator v0.11.2
   Compiling alloc-stdlib v0.2.2
   Compiling unic-ucd-ident v0.9.0
   Compiling parking_lot_core v0.9.10
   Compiling phf_macros v0.11.2
   Compiling brotli-decompressor v4.0.1
   Compiling toml_edit v0.20.2
   Compiling parking_lot v0.12.3
   Compiling same-file v1.0.6
   Compiling string_cache v0.8.7
   Compiling darling_macro v0.20.10
   Compiling phf v0.11.2
   Compiling walkdir v2.5.0
   Compiling smallvec v1.13.2
   Compiling brotli v6.0.0
   Compiling glob v0.3.1
   Compiling darling v0.20.10
   Compiling dunce v1.0.5
   Compiling serde_with_macros v3.9.0
   Compiling indexmap v1.9.3
   Compiling regex-automata v0.4.7
   Compiling swift-rs v1.0.7
   Compiling cssparser v0.27.2
   Compiling toml v0.8.2
   Compiling url v2.5.2
   Compiling uuid v1.10.0
   Compiling dyn-clone v1.0.17
   Compiling cfb v0.7.3
   Compiling html5ever v0.26.0
   Compiling jsonptr v0.4.7
   Compiling erased-serde v0.4.5
   Compiling regex v1.10.6
   Compiling bytes v1.7.2
   Compiling bitflags v2.6.0
   Compiling cargo_metadata v0.18.1
   Compiling urlpattern v0.3.0
   Compiling serde-untagged v0.1.6
   Compiling kuchikiki v0.8.2
   Compiling json-patch v2.0.0
   Compiling infer v0.16.0
   Compiling schemars v0.8.21
   Compiling serde_with v3.9.0
   Compiling option-ext v0.2.0
   Compiling powerfmt v0.2.0
   Compiling scopeguard v1.2.0
   Compiling futures-core v0.3.30
   Compiling deranged v0.3.11
   Compiling time-core v0.1.2
   Compiling pin-project-lite v0.2.14
   Compiling core-foundation-sys v0.8.7
   Compiling num-conv v0.1.0
   Compiling base64 v0.22.1
   Compiling core-foundation v0.10.0
   Compiling objc_exception v0.1.2
   Compiling anyhow v1.0.89
   Compiling http v1.1.0
   Compiling time v0.3.36
   Compiling toml_edit v0.19.15
   Compiling quick-xml v0.32.0
   Compiling tauri-utils v2.0.0-rc.12
   Compiling malloc_buf v0.0.6
   Compiling embed-resource v2.4.3
   Compiling objc v0.2.7
   Compiling core-graphics-types v0.2.0
   Compiling dirs-sys v0.4.1
   Compiling simd-adler32 v0.3.7
   Compiling adler2 v2.0.0
   Compiling foreign-types-shared v0.3.1
   Compiling heck v0.5.0
   Compiling miniz_oxide v0.8.0
   Compiling plist v1.7.0
   Compiling foreign-types v0.5.0
   Compiling dirs v5.0.1
   Compiling slab v0.4.9
   Compiling cargo_toml v0.17.2
   Compiling socket2 v0.5.7
   Compiling toml v0.7.8
   Compiling mio v1.0.2
   Compiling crc32fast v1.4.2
   Compiling tauri-winres v0.1.1
   Compiling pin-utils v0.1.0
   Compiling futures-task v0.3.30
   Compiling adler v1.0.2
   Compiling block v0.1.6
   Compiling futures-util v0.3.30
   Compiling miniz_oxide v0.7.4
   Compiling cocoa-foundation v0.2.0
   Compiling tokio v1.40.0
   Compiling flate2 v1.0.33
   Compiling core-graphics v0.24.0
   Compiling objc-sys v0.3.5
   Compiling fdeflate v0.3.4
   Compiling tauri-build v2.0.0-rc.12
   Compiling dpi v0.1.1
   Compiling objc2-encode v4.0.3
   Compiling raw-window-handle v0.6.2
   Compiling objc2 v0.5.2
   Compiling png v0.17.13
   Compiling cocoa v0.26.0
   Compiling crossbeam-utils v0.8.20
   Compiling http-body v1.0.1
   Compiling try-lock v0.2.5
   Compiling crossbeam-channel v0.5.13
   Compiling want v0.3.1
   Compiling pin-project v1.1.5
   Compiling objc2-foundation v0.2.2
   Compiling httparse v1.9.4
   Compiling semver v1.0.23
   Compiling futures-channel v0.3.30
   Compiling tracing-core v0.1.32
   Compiling tower-service v0.3.3
   Compiling tower-layer v0.3.3
   Compiling sha2 v0.10.8
   Compiling tower v0.4.13
   Compiling tracing v0.1.40
   Compiling objc2-app-kit v0.2.2
   Compiling hyper v1.4.1
   Compiling ico v0.3.0
   Compiling tauri-plugin v2.0.0-rc.12
   Compiling objc_id v0.1.1
   Compiling instant v0.1.13
   Compiling futures-sink v0.3.30
   Compiling dispatch v0.2.0
   Compiling unicode-segmentation v1.12.0
   Compiling tauri v2.0.0-rc.15
   Compiling lazy_static v1.5.0
   Compiling tokio-util v0.7.12
   Compiling tao v0.30.0
   Compiling keyboard-types v0.7.0
   Compiling tauri-plugin-shell v2.0.0-rc.3
   Compiling hyper-util v0.1.8
   Compiling mlua-sys v0.6.3
   Compiling tauri-codegen v2.0.0-rc.12
   Compiling wry v0.43.1
   Compiling http-body-util v0.1.2
   Compiling sync_wrapper v1.0.1
   Compiling serde_urlencoded v0.7.1
   Compiling mime v0.3.17
   Compiling ipnet v2.10.0
   Compiling serialize-to-javascript v0.1.1
   Compiling muda v0.14.1
   Compiling reqwest v0.12.7
   Compiling tauri-runtime v2.0.0-rc.12
   Compiling window-vibrancy v0.5.2
   Compiling tauri-runtime-wry v2.0.0-rc.13
   Compiling state v0.6.0
   Compiling pathdiff v0.2.1
   Compiling tauri-macros v2.0.0-rc.11
   Compiling embed_plist v1.2.2
   Compiling num-traits v0.2.19
   Compiling open v5.3.0
   Compiling poc v0.1.0 (/Users/weed/program language/obj/rust/poc/src-tauri)
   Compiling bstr v1.10.0
   Compiling shared_child v1.0.1
   Compiling os_pipe v1.2.1
   Compiling encoding_rs v0.8.34
   Compiling rustc-hash v2.0.0
   Compiling mlua v0.9.9
    Finished `dev` profile [unoptimized + debuginfo] target(s) in 41.92s
```
full info about `npm run tauri info`:
```shell
weed@MacBook-Pro src-tauri % npm run tauri info

> poc@0.1.0 tauri
> tauri info


[✔] Environment
    - OS: Mac OS 15.0.0 arm64 (X64)
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.79.0 (129f3b996 2024-06-10)
    ✔ cargo: 1.79.0 (ffa9cf99a 2024-06-03)
    ✔ rustup: 1.27.1 (2024-04-24)
    ✔ Rust toolchain: 1.79.0-aarch64-apple-darwin (default)
    - node: 22.6.0
    - pnpm: 9.0.6
    - yarn: 1.22.22
    - npm: 10.8.2

[-] Packages
    - tauri 🦀: 2.0.0-rc.15
    - tauri-build 🦀: 2.0.0-rc.12
    - wry 🦀: 0.43.1
    - tao 🦀: 0.30.0
    - tauri-cli 🦀: 1.5.14
    - @tauri-apps/api : 2.0.0-rc.5
    - @tauri-apps/cli : 2.0.0-rc.16

[-] Plugins
    - tauri-plugin-shell 🦀: 2.0.0-rc.3
    - @tauri-apps/plugin-shell : 2.0.0-rc.1

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../dist
    - devUrl: http://localhost:1420/
    - framework: Vue.js
    - bundler: Vite
```

--- a/.changes/fix-conditional-ios-deployment-target.md
+++ b/.changes/fix-conditional-ios-deployment-target.md
@@ -1,0 +1,6 @@
+---
+'tauri-cli': 'patch:bug'
+'@tauri-apps/cli': 'patch:bug'
+---
+
+The cli now only sets the iOS deployment target environment variable when building for iOS.

--- a/crates/tauri-cli/src/interface/rust.rs
+++ b/crates/tauri-cli/src/interface/rust.rs
@@ -151,10 +151,14 @@ impl Interface for Rust {
       std::env::set_var("MACOSX_DEPLOYMENT_TARGET", minimum_system_version);
     }
 
-    std::env::set_var(
-      "IPHONEOS_DEPLOYMENT_TARGET",
-      &config.bundle.ios.minimum_system_version,
-    );
+    if let Some(target) = &target {
+      if target.ends_with("ios") || target.ends_with("ios-sim") {
+        std::env::set_var(
+          "IPHONEOS_DEPLOYMENT_TARGET",
+          &config.bundle.ios.minimum_system_version,
+        );
+      }
+    }
 
     let app_settings = RustAppSettings::new(config, manifest, target)?;
 

